pkg/provider/metric_store.go

```diff
@@ -32,18 +32,33 @@ type externalMetricsStoredMetric struct {
 // MetricStore is a simple in-memory Metrics Store for HPA metrics.
 type MetricStore struct {
 	// metricName -> referencedResource -> objectNamespace -> objectName -> metric
```

Comment:
I think this comment is wrong/less valuable now. It's missing the labelsHash and could maybe be dropped if the type changes make this part clear now.